docs/guides/server/db.adoc

```diff
@@ -358,6 +357,136 @@ In the same way the migrationExport to point to a specific file and location:
 
 For more information, check the link:{upgrading_guide_link}#_migrate_db[Migrating the database] documentation.
 
+== Configure multiple datasources
+
+{project_name} allows you to specify multiple datasources in the case you need to access your own databases where your data lives.
+It means that you can connect to these databases to handle your data and execute required actions.
+It is very useful in situations where you have a database of your own users with a custom schema, and you cannot migrate to the {project_name} database due to multiple reasons.
+
+You can find more details on how to connect to your own users database in the link:{server_developer_guide}#_user-storage-spi[User Storage SPI] documentation.
+
+Defining multiple datasources works like defining a single datasource, with one important change - you have to specify a name (configuration property) for each datasource.
+
+=== Required configuration
+
+In order to enable an additional datasource, you need to set up 2 things - the JPA `persistence.xml` file and {project_name} configuration.
```

Comment:
I'm not sure if, but it might be good to mention that using a persistence.xml file cuts off the usage of quarkus orm properties: https://github.com/quarkusio/quarkus/issues/43216

This could come up if anyone is wanting the ability to dynamically control via configuration properties what goes into the persistence.xml file.

Reply:
@shawkins Added a note: https://github.com/keycloak/keycloak/pull/42655/commits/8ee16060ec3ccd34e6acb7b632c1b2db6bc6f854